node/network/availability-distribution/src/requester/mod.rs

```diff
@@ -207,16 +206,21 @@ impl Requester {
 					let tx = self.tx.clone();
 					let metrics = self.metrics.clone();
 
+					// We need to run the fetching task in the context of the leaf session,
+					// so we need to pass in the parent of the leaf to `with_session_info()`.
+					let leaf_parent =
+						get_block_ancestors(ctx, leaf, 1).await?.into_iter().next().unwrap_or(leaf);
```

Comment:
this will fetch block header for leaf and should always succeed, might make sense to issue a warning if it doesn't (`unwrap_or`)

Reply:
done